### PR TITLE
Rename `BaseArchUnitTestSuite#importedClasses` to `productionClasses`

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -227,7 +227,7 @@ jobs:
                   git tag "$tag"
                   git push origin "$tag"
             - run:
-                name: Create GitHub release
+                name: Create GitHub release & close milestone
```
Comment (PawelLipski) [inline .circleci/config.yml:230]: 🐷gyback

File: src/test/java/com/virtuslab/archunit/BackgroundTaskEnqueuingTestSuite.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ private boolean isTaskBackgroundableQueue(JavaMethodCall call) {
         })
         .because("otherwise it's likely that you forgot about actually scheduling the task; " +
             "mark the method with ${BackgroundableQueuedElsewhere.class.getSimpleName()} if this is expected")
-        .check(importedClasses);
```
Comment (PawelLipski) [inline src/test/java/com/virtuslab/archunit/BackgroundTaskEnqueuingTestSuite.java:61]: `imported` didn't really bring any meaning: if the classes is visible to ArchUnit, it's surely been imported somehow. `production`, however, states that the checks aren't executed against test classes.

File: .gitignore [hunk 3]
```diff
@@ -1,9 +1,11 @@
 build/
 *.class
 *.crt
+.DS_Store
```
Comment (PawelLipski) [inline .gitignore:4]: I've excluded the project folder from Spotlight indexing... but still, Spotlight probably managed to create these files right before I did so

File: .gitignore [hunk 4]
```diff
@@ -1,9 +1,11 @@
...
 **/.gradle/
 /gradlew.bat
 /*.hprof
 /*.hprof.*/
 /.idea/
 *.pem
+.sdkmanrc
```
Comment (PawelLipski) [inline .gitignore:11]: SDKMAN RiGCz

**Review by mkondratek — APPROVED** [review]
lgtm

**Pull request merged**